lib/models/User/index.js

```diff
@@ -0,0 +1,34 @@
+const userSchema = require('./schema.js');
+const mongoose = require('mongoose');
+
+mongoose.Promise = global.Promise;
+
+userSchema.statics.findOrCreate = findOrCreate;
+
+
+module.exports = mongoose.model('User', userSchema);
+
+
+function findOrCreate(newUser, done) {
+	this.findOne({ 'github.id': newUser.github.id })
+	.then((user) => {
+		if (!user) {
+			return this.create(newUser);
+		}
+
+		done(null, user);
+	})
+	.then((user) => {
+		done(null, user);
+	})
+	.catch((err) => {
+		done(err.message, null);
+	});
+}
+
+function findOrCreate2(newUser, done) {
```

Comment:
Does this function replaces findOrCreate? Because it seems like it does the same but written in a better way